### PR TITLE
docs(new): add TSDoc comments to `WebWorker`

### PR DESCRIPTION
**Pull request by AVGP**

### CODE REVIEW

File: src/WebWorker.ts [hunk 1]
```diff
@@ -30,12 +30,33 @@ type ExceptionThrownCallback = (
 ) => void;
 type JSHandleFactory = (obj: Protocol.Runtime.RemoteObject) => JSHandle;
 
+/**
+ * The WebWorker class represents a [WebWorker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API).
```
Comment (jackfranklin) [inline src/WebWorker.ts]: these need to use the TSDoc @link syntax: https://api-extractor.com/pages/tsdoc/tag_link/

File: src/WebWorker.ts [hunk 2]
```diff
@@ -30,12 +30,33 @@ type ExceptionThrownCallback = (
...
+ * The events `workercreated` and `workerdestroyed` are emitted on the page object to signal the worker lifecycle.
```
Comment (jackfranklin) [inline src/WebWorker.ts:37]: before this line you probably want to use a `@remarks` to split the summary from the examples, etc: https://api-extractor.com/pages/tsdoc/tag_remarks/

File: src/WebWorker.ts [hunk 3]
```diff
@@ -76,14 +97,31 @@ export class WebWorker extends EventEmitter {
     );
   }
 
+  /**
+   * Returns the URL for the script this web worker is running.
```
Comment (jackfranklin) [inline src/WebWorker.ts]: I think in this case we could lose the description in favour of just the `@returns ...` line rather than duplicate them both

**Review by jackfranklin — COMMENTED** [review]
@AVGP thank you so much for this! Few minor tweaks then I think this is good to go 👍

**Comment by AVGP** [comment]
@jackfranklin Thanks for the review! Slowly getting the hang of this :)

File: src/WebWorker.ts [hunk 4]
```diff
@@ -30,12 +30,35 @@ type ExceptionThrownCallback = (
 ) => void;
 type JSHandleFactory = (obj: Protocol.Runtime.RemoteObject) => JSHandle;
 
+/**
+ * The WebWorker class represents a {@link https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API | WebWorker}.
+ * The events `workercreated` and `workerdestroyed` are emitted on the page object to signal the worker lifecycle.
+ *
+ * @remarks
```
Comment (jackfranklin) [inline src/WebWorker.ts:36]: nitpick: I'd move the `@remarks` before the sentence about the events. The summary section (the bit before the `@remarks`) should be one short sentence at most.

**Comment by jackfranklin** [comment]
@AVGP one small comment then I think we can ship this!

You need to also run `npm run generate-docs` and commit the diff.

**Comment by AVGP** [comment]
@jackfranklin done :)

**Review by jackfranklin — APPROVED** [review]

**Pull request merged**